src/uct/base/uct_iface.h

```diff
@@ -155,7 +155,7 @@ enum {
  * Check the condition and return status as a pointer if not true.
  */
 #define UCT_CHECK_PARAM_PTR(_condition, _err_message, ...) \
-    if (ENABLE_PARAMS_CHECK && !(_condition)) { \
+    if (UCS_PARAM_CHECK_FAST_PATH(!(_condition))) { \
```

Comment:
IMO better make the names consistent, for example
UCS_PARAM_CHECK_FAST_PATH --> UCS_CHECK_PARAM
or
UCT_CHECK_PARAM -> UCT_PARAM_CHECK

Reply:
Regarding "UCS_PARAM_CHECK_FAST_PATH --> UCS_CHECK_PARAM", I think it should be 
UCS_PARAM_CHECK_FAST_PATH --> UCS_PARAM_CHECK because the name of the file is `param.h` under UCS.

The `_FAST_PATH` is actually leftovers. I thought to move `UCT_CHECK_PARAM` to `param.h` and call it `UCS_PARAM_CHECK`, This macro can replace many if statements in our code. I also thought to create another macro similar to this one without error message. It can also replace many if statements in our code.

WDYT?

Reply:
let's not expand this PR, which is mainly intended to fix performance issue and not refactor
we can rename UCS_PARAM_CHECK_FAST_PATH -> UCS_PARAM_CHECK if it doesn't affect many places